### ISSUE TITLE
Deno support

### ISSUE THREAD

**Issue by BlueskyFR** [nuxt/nuxt#29939]
### Describe the feature

Hi! It would be super nice to be able to develop with/install Nuxt using Deno, as the only listed options so far are npm, pnpm, yarn & bun 😊

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [x] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [x] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by typed-sigterm** [comment]
Do you mean add Deno starter to docs? Nuxt is compatible with Deno.

**Comment by web-dev-sam** [comment]
He probably meant Nuxi package manager options when creating a new project. When selecting npm and then trying to run `deno task build` it throws these errors a bunch:
 ![Image](https://github.com/user-attachments/assets/63704195-1afa-489d-b6c2-77c86387e63f)

**Comment by BlueskyFR** [comment]
Yes I meant that! And also to use deno instead of npm to install dependencies, so to add abother option "deno" alongside bun, nom, yarn and others.
Because I don't want to have to install npm if I already have deno.
And I am getting the same error yes!

**Comment by web-dev-sam** [comment]
After some source code delving I found out you can build nuxt using Deno by adding `nitro: { preset: 'deno' }` to your nuxt config. However I did not find a way to make it work for the development server so that it works. I assume that there isn't support for that yet. The issue could be related to some `fs.stat` or `fs.statSync` not receiving proper arguments ([deno-node polyfill](https://github.com/denoland/deno/blob/73411bb98a677727799122ebc397d825bf95b812/ext/node/polyfills/_fs/_fs_stat.ts#L434-L435)). But since the stacktrace is too small I don't know much more.

**Comment by BlueskyFR** [comment]
Is https://github.com/nuxt/cli/pull/526 solving this issue?
Gotta confirm that

**Comment by BlueskyFR** [comment]
Well when using the Nuxt UI starter with Deno I am getting a 500 server error for now:
```
[nuxt] [request error] [unhandled] [500] Cannot find module 'vue'
```

**Comment by danielroe** [comment]
Does it help if you add `vue` as a dependency?

**Comment by danielroe** [comment]
Deno seems to work fine in a new project with the latest CLI changes.

I checked UI template and installing `vue` does resolve the issue, so I added it to `package.json`.

**Comment by BlueskyFR** [comment]
@danielroe can you please reopen? This does not look fixed to me.

When creating a new project using `deno run -A npm:nuxi@latest init -t ui`, and then `deno task dev` I get this:
![Image](https://github.com/user-attachments/assets/df0fba72-499c-4f92-8443-ee464954fe93)
instead of (same project but using Yarn instead of Deno):
![Image](https://github.com/user-attachments/assets/8e9a27df-0bb2-498e-8b27-f6f5fe5592e4)

So I think this issue should still be opened 😅

**Comment by danielroe** [comment]
you're right, there's an issue:

![Image](https://github.com/user-attachments/assets/a9b90c86-542a-48f3-a884-ee421ca10395)

would you raise it upstream with deno? this doesn't seem like a nuxt bug to me.

**Comment by pi0** [comment]
Probably related (https://github.com/denoland/deno/issues/26700, https://github.com/denoland/deno/pull/23289)

**Comment by BlueskyFR** [comment]
@danielroe how are you getting that? I don't have any error on my side:![Image](https://github.com/user-attachments/assets/19252700-53c3-447a-9992-2a0ff7a7113e)

**Comment by BlueskyFR** [comment]
Are you sure you are using Deno 2.1.4? I think your issue might be already fixed

**Comment by danielroe** [comment]
quite right, this computer still had v2.0.6
